NAudio/Wave/SampleProviders/VolumeSampleProvider.cs

```diff
@@ -46,5 +55,53 @@ public int Read(float[] buffer, int offset, int sampleCount)
         /// Allows adjusting the volume, 1.0f = full volume
         /// </summary>
         public float Volume { get; set; }
+
+#if NETCOREAPP3_0
+        private unsafe void FastPath(float[] buffer, int offset, int sampleCount)
+        {
+            fixed (float* b = buffer)
+            {
+                var pStart = b + offset;
+                var pCurrent = pStart;
+                var pEnd = pStart;
+
+                if (Avx.IsSupported)
+                {
+                    var volume = Vector256.Create(Volume);
+                    var vector256SampleCount = sampleCount & ~7;
+                    pEnd = pStart + vector256SampleCount;
+                    while (pCurrent < pEnd)
+                    {
+                        var input = Avx.LoadVector256(pCurrent);
+                        var output = Avx.Multiply(input, volume);
+                        Avx.Store(pCurrent, output);
+                        pCurrent += 8;
+                    }
+                }
+
+                if (Sse.IsSupported)

```

Comment:
Maybe `else if (Sse.IsSupported)`?

Reply:
Current logic works optimally I'd say and is clean to read & easy to extend later.

Using edge case of 1005 samples:
On modern processors: 125 AVX, 1 SSE, 1 scalar. 
(With your suggestion, 125 AVX, 0 SSE, 5 scalar)
On old processors: 251 SSE, 1 scalar.
On arm processors: 1005 scalar.